### PR TITLE
Allow using a custom external `httpx` client

### PR DESCRIPTION
**Pull request by GuyKh**

### CODE REVIEW

File: glances_api/__init__.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ def __init__(
         ssl=False,
         username=None,
         password=None,
+        custom_httpx_client=None,
```
Comment (MartinHjelmare) [inline glances_api/__init__.py:24]: I'd just call this `httpx_client`.
```suggestion
        httpx_client=None,
```

File: glances_api/__init__.py [hunk 2]
```diff
@@ -31,13 +32,16 @@ def __init__(
         self.values = None
         self.username = username
         self.password = password
+        self.custom_httpx_client = custom_httpx_client
 
     async def get_data(self, endpoint):
         """Retrieve the data."""
         url = "{}/{}".format(self.url, endpoint)
 
+        httpxClient = this.custom_httpx_client if this.custom_httpx_client else httpx.AsyncClient()
```
Comment (MartinHjelmare) [inline glances_api/__init__.py:41]: Python prefers snake_case in variable names.
```suggestion
        httpx_client = self.httpx_client if self.httpx_client else httpx.AsyncClient()
```

File: glances_api/__init__.py [hunk 3]
```diff
@@ -31,13 +32,16 @@ def __init__(
         self.values = None
         self.username = username
         self.password = password
+        self.custom_httpx_client = custom_httpx_client
```
Comment (MartinHjelmare) [inline glances_api/__init__.py:35]:
```suggestion
        self.httpx_client = httpx_client
```

File: glances_api/__init__.py [hunk 4]
```diff
@@ -31,13 +32,16 @@ def __init__(
...
 
     async def get_data(self, endpoint):
         """Retrieve the data."""
         url = "{}/{}".format(self.url, endpoint)
 
+        httpxClient = this.custom_httpx_client if this.custom_httpx_client else httpx.AsyncClient()
+        
         try:
-            async with httpx.AsyncClient() as client:
+            async with httpxClient as client:
```
Comment (MartinHjelmare) [inline glances_api/__init__.py:44]:
```suggestion
            async with httpx_client as client:
```

**Comment by GuyKh** [comment]
@fabaff  -- please review

**Comment by fabaff** [comment]
#4 required changes. Thus, this PR needs to be updated.

File: glances_api/__init__.py [hunk 5]
```diff
@@ -33,13 +34,16 @@ def __init__(
         self.username = username
         self.password = password
         self.verify_ssl = verify_ssl
+        self.httpx_client = httpx_client
 
     async def get_data(self, endpoint):
         """Retrieve the data."""
         url = "{}/{}".format(self.url, endpoint)
 
+        httpx_client = self.httpx_client if self.httpx_client else httpx.AsyncClient(verify=self.verify_ssl)
```
Comment (GuyKh) [inline glances_api/__init__.py:43]: @fabaff - note that you're creating a new instance of httpx Async Client on each call, is this intentional?

Comment (fabaff) [reply glances_api/__init__.py:43]: I can't remember why I did it that way. Could be that i simple used the same approach for httpx that was there before.

Comment (GuyKh) [reply glances_api/__init__.py:43]: I'm not  sure if this should be reinitiated each time or can be reused.

I believe that the only problem in reusing the client one time is if there's an exception in the method or somehow the client is closed, the `httpx_client` won't be reinitialized (as far as I understand python behavior from being a Java developer 😕 ).

**Review by fabaff — APPROVED** [review]
Let's go with this and address the possible reuse of the client later.

**Pull request merged**